lib/order.rb

```diff
@@ -0,0 +1,91 @@
+require 'csv'
+require 'customer'
+
+class Order
+  VALID_STATUSES = [:pending, :paid, :processing, :shipped, :complete]
+  #adding a constant with the list of default fulfilment_status values
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status=:pending) #setting default value of status
+    validate_status(fulfillment_status) #calling validate_status method to check if a passed value is valid
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+  end
+
+  def total
+    array_costs = @products.values
+    return (array_costs.sum * 1.075).round(2)
+  end
+
+  #{ "banana" => 1.99, "cracker" => 3.00 }
+  def add_product(product_name, price)
+    @products.each_key do |key|
+      if product_name == key
+        raise ArgumentError.new
+      end
+    end
```

Comment:
You can use `include?` to simplify this.  Also, it's helpful to include the bad argument in argument errors:
```suggestion
    if @products.include?(product_name)
      raise ArgumentError.new "#{product_name} cannot be added because there is already a product with that name."
    end
```